Extract content from publications and organize it

tekrajchhetri opened the pull request:
This pull request implements the following feature.

- Extract the pdf content using grobid
- Organize the extracted content with headings to be used later for downstream tasks like NER.

tekrajchhetri committed:
tests added

tekrajchhetri committed:
example notebook and pdf publications (open access) added

tekrajchhetri committed:
dependencies updated

tekrajchhetri committed:
removed unused script

tekrajchhetri committed:
main code added to extract the pdf content

tekrajchhetri committed:
readme updated

tekrajchhetri committed:
readme app import instructions updated

tekrajchhetri committed:
correct notebook added

tekrajchhetri committed:
license info updated

tekrajchhetri committed:
pyproject toml and license txt updated to include MIT license

tekrajchhetri committed:
example code updated

tekrajchhetri committed:
error description added + docker command updated

tekrajchhetri committed:
cli command updated.

tekrajchhetri committed:
instruction updated to reflect cli change

tekrajchhetri committed:
example.ipynb to include cli example

tekrajchhetri committed:
Delete src/YOUR_PACKAGE_NAME directory

djarecka commented:
@tekrajchhetri - a quick question, are you able yo run the test?

tekrajchhetri commented:
I believe it was running file last time, will check it again.

tekrajchhetri committed:
updated readme

tekrajchhetri committed:
link to notebook added

kabilar commented on pyproject.toml:
```diff
@@ -1,30 +1,34 @@
-documentation = "https://YOUR_GITHUB_NICKNAME.github.io/YOUR_PACKAGE_NAME"
+homepage = "https://github.com/sensein/GrobidArticleExtractor"
+repository = "https://github.com/sensein/GrobidArticleExtractor"
+documentation = "https://tekrajchhetri.github.io/GrobidArticleExtractor"
```
This site doesn't seem to resolve.

kabilar commented on examples/example.ipynb:
I would suggest simplifying this notebook by installing directly from PyPI and removing the lines that have errors.

kabilar replied:
Perhaps lets rename this notebook to `getting-started.ipynb`?

tekrajchhetri replied:
sure

kabilar commented on examples/pdf_files/gpt-ner.pdf:
Do you think that arXiv would be okay with us storing this file publicly in this repo?

tekrajchhetri replied:
@kabilar  It's open access so I do not see any issue. I might be wrong, though! If you think there would be an issue, I can remove it.

kabilar commented on src/GrobidArticleExtractor/cli.py:
```diff
@@ -0,0 +1,90 @@
+# software or the use or other dealings in the software.
+# -----------------------------------------------------------------------------
+
+# @Author  : Tek Raj Chhetri
+# @Email   : tekraj@mit.edu
+# @Web     : https://tekrajchhetri.com/
+# @Software: PyCharm
+
```
Perhaps this is not needed since it is in the `pyproject.toml`?

```suggestion
```

kabilar commented on src/GrobidArticleExtractor/cli.py:
```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# DISCLAIMER: This software is provided "as is" without any warranty,
+# express or implied, including but not limited to the warranties of
+# merchantability, fitness for a particular purpose, and non-infringement.
+#
+# In no event shall the authors or copyright holders be liable for any
+# claim, damages, or other liability, whether in an action of contract,
+# tort, or otherwise, arising from, out of, or in connection with the
+# software or the use or other dealings in the software.
+# -----------------------------------------------------------------------------
+
```
Perhaps this is not needed since it is in the `LICENSE` file?

```suggestion
```

kabilar commented on src/GrobidArticleExtractor/app.py:
```diff
@@ -0,0 +1,306 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# DISCLAIMER: This software is provided "as is" without any warranty,
+# express or implied, including but not limited to the warranties of
+# merchantability, fitness for a particular purpose, and non-infringement.
+#
+# In no event shall the authors or copyright holders be liable for any
+# claim, damages, or other liability, whether in an action of contract,
+# tort, or otherwise, arising from, out of, or in connection with the
+# software or the use or other dealings in the software.
+# -----------------------------------------------------------------------------
```
Perhaps this is not needed since it is in the `LICENSE` file?

```suggestion
```

kabilar commented on src/GrobidArticleExtractor/app.py:
```diff
@@ -0,0 +1,306 @@
+# software or the use or other dealings in the software.
+# -----------------------------------------------------------------------------
+
+# @Author  : Tek Raj Chhetri
+# @Email   : tekraj@mit.edu
+# @Web     : https://tekrajchhetri.com/
+# @Software: PyCharm
+
```
Perhaps this is not needed since it is in the `pyproject.toml`?

```suggestion
```

tekrajchhetri committed:
renamed notebook name

tekrajchhetri committed:
Update src/GrobidArticleExtractor/cli.py

Co-authored-by: Kabilar Gunalan <kabilar.gunalan@gmail.com>

tekrajchhetri committed:
Update src/GrobidArticleExtractor/app.py

Co-authored-by: Kabilar Gunalan <kabilar.gunalan@gmail.com>

tekrajchhetri committed:
Update src/GrobidArticleExtractor/app.py

this comment is automatically generated for each python file because of how i've configured my IDE

Co-authored-by: Kabilar Gunalan <kabilar.gunalan@gmail.com>

tekrajchhetri committed:
Update src/GrobidArticleExtractor/cli.py

this comment is automatically generated for each python file because of how i've configured my IDE

Co-authored-by: Kabilar Gunalan <kabilar.gunalan@gmail.com>
